src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,107 @@
-package core.basesyntax;
-
-public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
-
-    }
-
-    @Override
-    public void add(T value, int index) {
-
-    }
-
-    @Override
-    public void addAll(List<T> list) {
-
-    }
-
-    @Override
-    public T get(int index) {
-        return null;
-    }
-
-    @Override
-    public void set(T value, int index) {
-
-    }
-
-    @Override
-    public T remove(int index) {
-        return null;
-    }
-
-    @Override
-    public T remove(T element) {
-        return null;
-    }
-
-    @Override
-    public int size() {
-        return 0;
-    }
-
-    @Override
-    public boolean isEmpty() {
-        return false;
-    }
-}
+package core.basesyntax;
+
+import java.util.NoSuchElementException;
+
+public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int GROW_FACTORY = DEFAULT_CAPACITY >> 1;
+    private int size = 0;

```

Comment:
```suggestion
    private int size;
    
```

0 is default value of int